### DISCUSSION TITLE
Docker issue: ROOT level privileges prohibited!

### DISCUSSION THREAD

**Discussion by TH-Braemer** [abraunegg/onedrive#1310] [Q&A] [answered]
Hi again.

I'm sorry to say, that I think you are a bit to quick closing issues.
We are now two (@BelphegorPrime) facing the same issue with this container. I have trawled Google my very best, but I'm still not able to make it work.
I'm using Ubintu Server 20.04.2 LTS and I have about 15 other containers running without any issues, so I'm not just gonna switch OS.
It might be right that it appears that I'm trying to run the container as root, but it's definitely not by intension.
Telling us to look at Google doesn't really help... You might need to be more specific.

@BelphegorPrime if you have any luck getting it up and running - please let me know.

Second @abraunegg, I don't understand why you recommend your own container to me in other repository discussions, when I have already mentioned for you, that I can't make it work...

**Comment by abraunegg** [comment, marked as answer]
@TH-Braemer 
Suggest that you step through each and every step from start to finish on *how* you are downloading & starting the Docker container.

Some simple questions to think about:
* Am I logging into my system as a normal, unprivileged user or as root?
* Have I correctly configured Docker so that I can start containers as my unprivileged user without the use of `sudo` - or do I have to use sudo as thats the only way it works?
* How does my entire process / download / configure / run - compare to each command as specified in the docker readme? What differences are there between the commands I am running and the commands as documented which *have* been verified as accurate and true
* Am I doing any weird mounting of external devices that are the storage points that I am trying to use with the Docker container?
* Have I needed to make any changes to my SELinux configuration because I am trying to mount something in an odd place?

I 100% suspect that your issue is localised to something that you are doing on your system.

The docker readme contains exact details that have been verified on *multiple* systems. Why it is not working for you - unsure - but potentially could be with how you are doing things. 

Start from scratch, follow the instructions as per the document and see where that leads you.

I am away from the PC atm, but tomorrow I can attempt to replicate  Ubuntu Server platform with Docker and show you step-by-step what is done ...

**Reply by TH-Braemer** [reply]
I can say check to all your suggestions but one...
The only thing I might not have tried is to mount folders directly in the "host" filesystem.
Right now the download folder points to a folder mounted (in fstab) to a folder on an external USB-drive.

I will try to see if I can get the time to try that tonight...

**Reply by TH-Braemer** [reply]
So I tried again, but this time I used a folders in the "host" filesystem for my config and data. Folders owned be my user (not root)
That helped - Now It seems to be running as expected :)

Thank you for the hints - I believe I'm Up'N'Running now 👍 

@BelphegorPrime - I don't know if this might help you too?

**Reply by abraunegg** [reply]
@TH-Braemer 
Glad you are up and running

**Reply by abraunegg** [reply]
@TH-Braemer 
Additionally, given your issue appeared to be where your config & data folders are stored, as per the [Docker.md](https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md#2-prepare-config-volume) file:

=======

Onedrive needs two volumes. One of them is the config volume. Create it with:

```bash
docker volume create onedrive_conf
```

This will create a docker volume labeled `onedrive_conf`, where all configuration of your onedrive account will be stored. You can add a custom config file and other things later.

The second docker volume is for your data folder and is created in the next step. It needs the path to a folder on your filesystem that you want to keep in sync with OneDrive. Keep in mind that:

-   The owner of your specified folder must not be root
-   The owner of your specified folder must have permissions for its parent directory

=======

Was it the 'config' volume or 'data' volume which was giving you the problem?

**Reply by BelphegorPrime** [reply]
@TH-Braemer I already found rclone for my usecase. I wanted some folder where I could drop files that will only be uploaded....

EDIT:
I would have had the same problem with the filesystem, at least I think, I planned also to mount the data folder on a mounted RAID on my NAS...

**Reply by TH-Braemer** [reply]
@abraunegg 
I must admit, that at first I wasn't aware of the fact that a NTFS mounted USB-drive always "state" root ownership for the folders - even though i actually isn't. That was my issue!
I'm not directly creating docker volumes for 'config' and 'data' since I'm using docker-compose.
I changed my setup to use two folders located in the filesystem on the computer, not a mounted drive.
For the account-setup I created to variables:
`onedriveConfig="${HOME}/docker/OneDrive/Config"`
`onedriveData="${HOME}/docker/OneDrive/Backup"`
And then started a temporary container:
`docker run --rm -it --name onedrive -v "${onedriveConfig}:/onedrive/conf" -v "${onedriveData}:/onedrive/data" -e ONEDRIVE_UID:1000 -e ONEDRIVE_GID:1000 -e ONEDRIVE_DOWNLOADONLY:1 driveone/onedrive`
When the sync started, I stopped the container, and started it with my docker-compose file.

Why am I using mounted USB-drive?
I have made a setup, where all my containers are spawned from docker-compose files. These files I have on my Docker host server, which is a plain Ubuntu Server 20.04.2. 
All containers are configures so that any necessary folders are located/mounted on the USB-drive.
By doing this, I can in a matter of minutes move all my docker "servers" to a different docker host computer. All I need is the docker-compose files, and the USB-drive.
One of my containers are Home-Assistant controlling light and other stuff in our house.
I once had a mishap updating my Ubuntu server, and in 5 minutes I had it all running on a different computer.
Trust my - my wife is not very tolerant, when it comes to "down-time" in the house control...

I will try to make a Linux formatted  USB-drive mounted it in fstab to see if I can continue this setup...

**Reply by abraunegg** [reply]
@TH-Braemer 
Please can you review the updated [docker readme](https://github.com/abraunegg/onedrive/blob/master/docs/Docker.md) - which I have validated on Ubuntu Server 20.04 + RHEL 8

**Reply by TH-Braemer** [reply]
@abraunegg 
I think it looks very good.
I'm sure the section about the mount might save others from the issues I had 👍
